### PR TITLE
2b endpoints swb

### PR DESCRIPTION
**Pull request by Scott-Borecki**
### What I Did:
- Create endpoints from database resources:
  - `GET /api/v1/gyms/:id`
  - `POST /api/v1/users/:user_id/gyms/:gym_id/gym_members`
  - `DELETE /api/v1/users/:user_id/gyms/:gym_id`
- Add edge case testing to `POST /api/v1/users/:user_id/gyms/:gym_id/events`
- Set up test framework for:
  - `GET /api/v1/users/:user_id/gyms/:gym_id/events/new`
  - `POST /api/v1/users/:user_id/friendships`

### Road Blocks
- None

### Test Coverage
- [X] I have tested this code with RSpec: 
  - All tests passing.
  - 141/142 test coverage (we have not used the `json_error_response` method yet)
- The serializers were created with the `json-apiserializer` gem; however, we should consider adding tests for the serializers.
- Validations need to be added to endpoint request edge cases and tested.

### Questions/Notes:
- We should add a pull request template :)

### CODE REVIEW

**Review by tvaroglu — APPROVED** [review]
Nice work, love the way you are returning the errors for bad requests, and setting up conditionals within the controller actions.  Super clean and easy to follow.

File: config/routes.rb [hunk 1]
```diff
@@ -3,13 +3,18 @@
 
   namespace :api do
     namespace :v1 do
+      delete 'users/:user_id/gyms/:gym_id', to: 'users/gyms/gym_members#destroy'
+
```
Comment (Scott-Borecki) [inline config/routes.rb:7]: @tvaroglu @bfl3tch 

Thinking about this one further, if we want to be more RESTful, we could put this route in an Api::V1::GymMembersController with a `destroy` action and take the `user_id` and `gym_id` as query params.  So the main difference would be getting the ids from query params instead of from the path.  Aside from being more RESTful, it comes down to how we want to call it on the FE.

Same thought process could apply to `users/gyms/events#create` and `users/gyms/events#create`.

Let me know what you all think!

Comment (Scott-Borecki) [reply config/routes.rb:7]: One other thing to consider is security.  We don't want a malicious user to be able to delete records when they shouldn't be deleted.  But maybe that's another discussion :)

**Comment by bfl3tch** [comment]
Awesome work!
